src/base/QXmppUtils.cpp

```diff
@@ -322,3 +323,69 @@ void helperToXmlAddTextElement(QXmlStreamWriter *stream, const QString &name,
     else
         stream->writeEmptyElement(name);
 }
+
+/// \cond
+
+//
+// Generates a random count of random bytes.
+//
+// \param minimumByteCount minimum count of bytes to generate
+// \param maximumByteCount maximum count of bytes to generate
+//
+// \return the generated bytes
+//
+QByteArray QXmpp::Private::generateRandomBytes(uint32_t minimumByteCount, uint32_t maximumByteCount)
+{
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
+    const auto byteCount = QRandomGenerator::system()->bounded(minimumByteCount, maximumByteCount);
+#else
+    const auto byteCount = (qrand() % (maximumByteCount - minimumByteCount)) + minimumByteCount;
+#endif
+    QByteArray bytes;
+    bytes.resize(byteCount);
+    generateRandomBytes(reinterpret_cast<uint8_t *>(bytes.data()), byteCount);
+
+    return bytes;
+}
+
+//
+// Generates random bytes.
+//
+// \param bytes generated bytes
+// \param byteCount count of bytes to generate
+//
+void QXmpp::Private::generateRandomBytes(uint8_t *bytes, uint32_t byteCount)
+{
+    // QRandomGenerator does not provide a random generation of single bytes.  Thus, this approach
```

Comment:
After three iterations, we have a simple explanation, thanks :)
```suggestion
    // QRandomGenerator does not provide a random generation of single bytes. Thus, this approach
```